common/scripted_effects/wc_scheme_effects.txt

```diff
@@ -0,0 +1,186 @@
+﻿#Sets up the Outcome Roll values
+stasis_trap_outcome_roll_setup_effect = {
+	#Discovery Roll setup
+	save_scope_value_as = {
+		name = discovery_chance
+		value = {
+			value = 100
+			subtract = scope:scheme.scheme_secrecy
+		}
+	}
+}
+
+#Roll to check whether a Stasis Trap Scheme is a success or a failure, and whether the owner is discovered
+stasis_trap_outcome_roll_effect = {
+	#SUCCESS ROLL
+	random = {
+		chance = scope:scheme.scheme_success_chance
+		custom_tooltip = stasis_trap_successful_roll_tt
+		save_scope_value_as = {
+			name = scheme_successful
+			value = yes
+		}
+	}
+	### End Success roll
+	#DISCOVERY ROLL
+	custom_label = stasis_trap_success_discovery_tt
+	random = {
+		chance = scope:discovery_chance
+		custom_tooltip = stasis_trap_become_discovered_roll_tt
+		save_scope_value_as = {
+			name = scheme_discovered
+			value = yes
+		}
+	}
+	### End Discovery roll
+
+	hidden_effect = {
+		#FIRE CORRECT ON ACTIONS
+		if = {
+			limit = {
+				exists = scope:scheme_successful
+			}
+			trigger_event = {
+				on_action = stasis_trap_succeeded
+			}
+		}
+		else = {
+			trigger_event = {
+				on_action = stasis_trap_failed
+			}
+		}
+	}
+}
+
+
+#Used in the immediate of the Scheme Owner's success event
+stasis_trap_success_effect = {
+	
+	scope:target = {
+		add_character_flag = {
+			flag = was_stasis_traped_block_notification_event
```

Comment:
I looked for this flag. It's never checked. Something is wrong compared to the vanilla?